test/test_luaunit.lua

```diff
@@ -690,9 +690,13 @@ TestLuaUnitAssertions = { __class__ = 'TestLuaUnitAssertions' }
         lu.TABLE_EQUALS_KEYBYCONTENT = false
         assertFailure( lu.assertEquals, {[{}] = 1}, { [{}] = 1})
         assertFailure( lu.assertEquals, {[{one=1, two=2}] = 1}, { [{two=2, one=1}] = 1})
+        assertFailure( lu.assertEquals, {[{1}]=2, [{1}]=3}, {[{1}]=3, [{1}]=2} )
         lu.TABLE_EQUALS_KEYBYCONTENT = true
         lu.assertEquals( {[{}] = 1}, { [{}] = 1})
         lu.assertEquals( {[{one=1, two=2}] = 1}, { [{two=2, one=1}] = 1})
+        lu.assertEquals( {[{1}]=2, [{1}]=3}, {[{1}]=3, [{1}]=2} )
+        -- try the other order as well, in case pairs() returns items reversed in the test above
+        lu.assertEquals( {[{1}]=2, [{1}]=3}, {[{1}]=2, [{1}]=3} )
```

Comment:
Yes, as I commented earlier: things tend to turn ugly quickly with "table equality" matching of table-type keys.

But I think @AlfonZ42 has some valid point here: If we accept the concept of "equality" based on table content (`M.TABLE_EQUALS_KEYBYCONTENT = true`, dating back to #30), then the example in [his comment](https://github.com/bluebird75/luaunit/pull/57#issuecomment-226185977) makes sense. When supporting this, we probably need to go all the way...